### PR TITLE
attach file

### PR DESCRIPTION
**Pull request by jburel**
use method from gateway
Attach a file as a file annotation using the gateway method

cc @dominikl @pwalczysko

### CODE REVIEW

File: practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy [hunk 1]
```diff
@@ -298,48 +298,10 @@ def save_summary_as_csv(file, rows, columns) {
...
+    data = new DatasetData(new DatasetI(dataset_id, false))
     namespace = "training.demo"
-    fa = new FileAnnotationI()
-    fa.setFile(original_file)
-    fa.setNs(rstring(namespace))
-
-    data_object = new DatasetData(new DatasetI(dataset_id, false)) 
-    svc.attachAnnotation(ctx, new FileAnnotationData(fa), data_object)
+    mimetype = "text/csv"
+    svc.attachFile(ctx, file, mimetype, "", file.getName(), namespace, data)
```
Comment (dominikl) [inline practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy]: The `file.getName()` parameter is too much: Method signature is `attachFile(SecurityContext ctx, File file, String mimetype, String description, String namespace, DataObject target)`.

Comment (jburel) [reply practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy]: Copy paste from other code
thanks for checking

**Comment by jburel** [comment]
@dominikl fixed

File: practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy [hunk 2]
```diff
@@ -298,48 +298,10 @@ def save_summary_as_csv(file, rows, columns) {
...
+    data = new DatasetData(new DatasetI(dataset_id, false))
     namespace = "training.demo"
-    fa = new FileAnnotationI()
-    fa.setFile(original_file)
-    fa.setNs(rstring(namespace))
-
-    data_object = new DatasetData(new DatasetI(dataset_id, false)) 
-    svc.attachAnnotation(ctx, new FileAnnotationData(fa), data_object)
+    mimetype = "text/csv"
+    svc.attachFile(ctx, file, mimetype, "", namespace, data)
```
Comment (dominikl) [inline practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy]: Noticed one more issue. The attachFile method is a bit weird, it returns a Future object. And if you don't wait for it to finish (with `get()`) the chance is high that it won't actually create the file annotation.

```suggestion
    future = svc.attachFile(ctx, file, mimetype, "", namespace, data)
    future.get()
```

Comment (joshmoore) [reply practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy]: Because the script ends before it's complete?

Comment (jburel) [reply practical/groovy/analyse_dataset_save_rois_and_summary_table.groovy]: most likely

**Comment by dominikl** [comment]
👍  Tested against outreach, works fine. Good to merge.

**Comment by jburel** [comment]
Thanks @dominikl 
Merging so it can be used for next training

**Pull request merged**